### ISSUE TITLE
Support for cross compilation

### ISSUE THREAD

**Issue by noraj** [crystal-lang/shards#627]
`shards build` is more convenient than `crystal build` for building multiple binaries. However, `shards build` doesn't support `--cross-compile` and `--target` flags than are a must for static compilation since only Alpine is supported and so most people need to compile inside a docker.

**Comment by straight-shoota** [comment]
This is already supported. `shards build` delegates all options following the build target name to `crystal build`.
So indeed `shards build --cross-compile` invokes `crystal build ... --cross-compile`.

**Comment by straight-shoota** [comment]
`shards build` forwards all options it does not recognize. But `x86_64-linux-musl` is recognized as a target name. I think it's generally a good idea to write a single option with key and value joined by `=`.
Alternatively, you could also use double dash to tell `shards build` that the following options are to be forwarded: `shards build drift  --verbose -- --cross-compile --target x86_64-linux-musl`.

**Comment by straight-shoota** [comment]
It might be a good idea to change the forwarding behaviour so that *any* unrecognized option has the same effect as `--` and all following options won't be parsed as options of `shards build`.
This forces a certain order of options: all target names and options for `shards build` itself need to go before any forwarded options. This is probably what most users do and expect to happen.

**Comment by HertzDevil** [comment]
Related: crystal-lang/crystal#5845

**Comment by luislavena** [comment]
> Alternatively, you can also use double dash to tell `shards build` that the following options are to be forwarded: `shards build drift --verbose -- --cross-compile --target x86_64-linux-musl`.

Sadly, that is not how `build` has been coded, so that doesn't work:

```console
$ shards build drift --verbose -- --cross-compile --target x86_64-linux-musl
db: checking...
sqlite3: checking...
db: checking...
Dependencies are satisfied
Building: drift
crystal build -o /app/bin/drift src/cli.cr --verbose
```

None of the options after `--` are passed to `crystal build`.

Only `shards run` forwards options after `--`: https://github.com/crystal-lang/shards/blob/master/src/cli.cr#L66-L69

**Comment by straight-shoota** [comment]
Indeed. I had incorrectly inferred that to work. I'm pretty sure it *should* work. But it does not.

**Comment by ysbaddaden** [comment]
> I'm pretty sure it should work, though.

Yes, it definitely should.
